neutron_fwaas/tests/unit/services/firewall/service_drivers/test_driver_api.py

```diff
@@ -196,7 +200,51 @@ def test_update_firewall_policy(self):
                                        self.driver_api,
                                        payload=self.m_payload)
 
-    # Test Firewall Rule
+    def test_add_router_external_port_to_fwg(self):
+        self.plugin.l3_plugin = mock.Mock()
+        with self.network() as network:
```

Comment:
[optional/question] If you don't explicitly need to create the network I _think_ a `with self.subnet()` creates a new network implicitly. You could just use the tenant id of the subnet in your checks below. Just to save some indentantion / calls to `network()`. :)